ConfigSpace/io/pcs_new.py

```diff
@@ -0,0 +1,443 @@
+        except pyparsing.ParseException:
+            pass
+
+        try:
+            if "categorical" in line:
+                param_list = pp_cat_param.parseString(line)
+                name = param_list[0]
+                choices = [choice for choice in param_list[3:-4:2]]
+                default = param_list[-2]
+                param = create["categorical"](name=name, choices=choices, default=default)
+                cat_ct += 1
+                
+            elif "ordinal" in line:
+                param_list = pp_ord_param.parseString(line)
+                name = param_list[0]
+                sequence = [seq for seq in param_list[3:-4:2]]
+                default = param_list[-2]
+                param = create["ordinal"](name=name, sequence=sequence, default=default)
+                ord_ct += 1
+                
+        except pyparsing.ParseException:
+            pass
+
+        if param is None:
+            raise NotImplementedError("Could not parse: %s" % line)
+
+        configuration_space.add_hyperparameter(param)
+
+    for clause in forbidden:
+        param_list = pp_forbidden_clause.parseString(clause)
+        tmp_list = []
+        clause_list = []
+        for value in param_list[1:]:
+            if len(tmp_list) < 3:
+                tmp_list.append(value)
+            else:
+                # So far, only equals is supported by SMAC
+                if tmp_list[1] == '=':
+                    # TODO maybe add a check if the hyperparameter is
+                    # actually in the configuration space
+                    clause_list.append(ForbiddenEqualsClause(
+                        configuration_space.get_hyperparameter(tmp_list[0]),
+                        tmp_list[2]))
+                else:
+                    raise NotImplementedError()
+                tmp_list = []
+        configuration_space.add_forbidden_clause(ForbiddenAndConjunction(
+            *clause_list))
+            
+    conditions_per_child = OrderedDict()
+    for condition in conditions:
+        child_name = condition[0]
+        if child_name not in conditions_per_child:
+            conditions_per_child[child_name] = list()
+        conditions_per_child[child_name].append(condition)
+
+    for child_name in conditions_per_child:
+        for condition in conditions_per_child[child_name]:
+            condition = condition[2:]
+            condition = ' '.join(condition)
+            if '||' in str(condition):
+                ors = []
+                # 1st case we have a mixture of || and &&
+                if '&&' in str(condition):
+                    ors_combis = []
+                    for cond_parts in str(condition).split('||'):
+                        condition = str(cond_parts).split('&&')
+                        # if length is 1 it must be or
+                        if len(condition) == 1:
+                            element_list = [element for part in condition for element in part.split()]
```

Comment:
It converts the string which results from both splits (|| and &&) into a list of strings in order to process the condition type better. e.g. condition: ['classifier == k_nearest_neighbors '] ;  element_list = ['classifier', '==', 'k_nearest_neighbors']